mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -40,20 +40,29 @@ import org.apache.spark.util.random.XORShiftRandom
  * to it should be cached by the user.
  */
 @Since("0.8.0")
-class KMeans private (
+class KMeans @Since("2.3.0") private (
```

Comment:
If it's a private constructor, it shouldn't have API "Since" annotations. You don't need to preserve it for compatibility.